src/firefly/test/edu/caltech/ipac/astro/IpacTableReaderTest.java

```diff
@@ -0,0 +1,801 @@
+        |   deg     |    deg    |          |   mag    |            |
+        |   null    |    null   |   null   |   null   |   null     |
+          165.466279  -34.704730      5       11.27       K6Ve
+          123.4       5.67            9       8.9         K6Ve-1
+
+        */
+
+        //Inputs:
+        String tableName = "IPACTableSample1.tbl";
+        File inFile = new File(FileLoader.getDataPath(IpacTableReaderTest.class) + tableName);
+        //onlyColumns = null;
+        useFloatsForDoubles = true;
+        isHeadersOnlyAllow = true;
+
+        //Reset the expected values:
+        setExpectedValues_Input1_full();
+
+        //Read the table and generate the dataGroup:
+        dataGroup = IpacTableReader.readIpacTable(inFile, onlyColumns, useFloatsForDoubles, catName, isHeadersOnlyAllow);
+
+        //Check the result:
+        checkResult(dataGroup);
+
+        //Test result: No effects from  useFloatsForDoubles = true; isHeadersOnlyAllow = true;
+        // useFloatsForDoubles is not used. isHeadersOnlyAllow = true is not used.
+        // Passed.
+        //Action: Do we need to use those two parameters and for what?
+
+    }
+
+    @Test
+    /**
+     * This test calls the method IpacTableReader.readIpacTable to read only two columns from a valid input IPAC table, IPACTableSample1.tbl,
+     * and verify if the produced dataGroup contains the correct information from the table.
+     *
+     */
+    public void testOnlyColumns() throws IpacTableException, ClassNotFoundException, java.lang.NullPointerException
+
+    {
+
+        /* IPACTableSample1.tbl:
+
+        \catalog1 = 'Sample Catalog1'
+        \catalog2 = 'Sample Catalog2'
+        \ Comment1
+        \ Comment2
+        |   ra      |    dec    |   n_obs  |    V     |   SpType   |
+        |   double  |    double |   int    |   real   |   char     |
+        |   deg     |    deg    |          |   mag    |            |
+        |   null    |    null   |   null   |   null   |   null     |
+          165.466279  -34.704730      5       11.27       K6Ve
+          123.4       5.67            9       8.9         K6Ve-1
+
+        */
+
+
+        //Inputs:
+        String tableName = "IPACTableSample1.tbl";
+        File inFile = new File(FileLoader.getDataPath(IpacTableReaderTest.class) + tableName);
+        //The same input file but only read two columns:
+        onlyColumns = new String[]{"ra", "dec"};
+        useFloatsForDoubles = false;
+        isHeadersOnlyAllow = false;
+
+        //Reset the expected values:
+        setExpectedValues_Input1_radecOnly();
+
+        //Read the table and generate the dataGroup:
+        dataGroup = IpacTableReader.readIpacTable(inFile, onlyColumns, useFloatsForDoubles, catName, isHeadersOnlyAllow);
+
+        //Check the result:
+        checkResult(dataGroup);
+
+        //Test result: "onlyColumns" works fine. Only those two columns are read.
+        // Passed.
```

Comment:
I will use this test case to post an example of my prefer method:
For me, the example below is simple and easy to read.  The input and the expected results for this test case are all in one place. 
```
        String input =
        "\\catalog1 = 'Sample Catalog1'\n" +
        "\\catalog2 = 'Sample Catalog2'\n" +
        "\\ Comment1\n" +
        "\\ Comment2\n" +
        "|   ra      |    dec    |   n_obs  |    V     |   SpType   |\n" +
        "|   double  |    double |   int    |   real   |   char     |\n" +
        "|   deg     |    deg    |          |   mag    |            |\n" +
        "|   null    |    null   |   null   |   null   |   null     |\n" +
        " 165.466279  -34.704730      5       11.27       K6Ve       \n" +
        " 123.4       5.67            9       8.9         K6Ve-1     \n";

        DataGroup dg = IpacTableReader.readIpacTable(new StringReader(input), "onlyColumns",  new String[]{"ra", "dec"}, false, 0);

        Assert.assertEquals("There should be 2 columns", 2, dg.getDataDefinitions().length);
        Assert.assertEquals("There should be 2 rows", 2, dg.size());
        Assert.assertEquals("ra for row 1", 165.466279, (Double)dg.get(0).getDataElement("ra"), 0.001);
        Assert.assertEquals("dec for row 2", 5.67, (Double)dg.get(1).getDataElement("dec"), 0.001);
```